### DISCUSSION TITLE
Import problem from unit test file.

### DISCUSSION THREAD

**Discussion by DKnoto** [albilu/netbeansPython#31] [General]
After standard project generation, I added the func1 function to the main.py file.
I then generated a standard test unit file. After these operations, the unit tests worked.
Next, I wanted to test the func1 function. I added the following code:
`
from main import func1
`
Unfortunately, this operation fails because the PYTHONPATH environment variable 
contains the `/.../Project/tests` directory but not the `/.../Project/` directory.
I solved this problem by adding the appropriate PYTHONPATH variable in the global
configuration `NetBeans/Options/Python/Environment Variables` but this is not 
a good solution.

Can this be set somewhere just for a specific project? Maybe in the
 `nbproject/project.properties` file?

**Comment by albilu** [comment]
Indeed. This could be handle in a project specific env.json file instead. I prefer a json file instead of project.properties because i am not sure if it handles well multi values.

I will also look how vscode deal with this.

**Comment by DKnoto** [comment]
JSON files are good but when opened in NB completely unreadable, everything is on one line. It would be useful to have an equivalent of such presentation as in  [`jq`](https://stedolan.github.io/jq/)

**Reply by albilu** [reply]
Agree. i have another plugin that solve json problems on Netbeans that i will publish soon

**Comment by DKnoto** [comment]
Today I was testing a new small project in NB 19 and netbeansPython 1.5. The import problem in unit tests still occurs. Is it already possible to set the correct PYTHONPATH somewhere?

**Reply by albilu** [reply]
This feature request is still in the backlog #32 
I don't always have the time to work on this project. Feel free to support this project by sponsoring it. It will help prioritize your particular feature request
Thanks

**Reply by DKnoto** [reply]
I solved this problem by preparing a dedicated `Options/Python/Platforms` for the project. And for a given platform, I set the `PYTHONPATH` variable accordingly.
